pkgs/development/web/nodejs/v22.nix

```diff
@@ -52,11 +52,11 @@ buildNodejs {
       ./use-correct-env-in-tests.patch
       ./bin-sh-node-run-v22.patch
 
-      # Can be removed after https://github.com/NixOS/nixpkgs/pull/403958.
+      # Fix for flaky test
+      # TODO: remove when included in a release
       (fetchpatch2 {
-        url = "https://github.com/nodejs/node/commit/9aa57bf8dab2dbfb8b6974fe71d5dbe6daf66244.patch?full_index=1";
-        hash = "sha256-k3h8mPgvaIYGAkGmaL+ix7kUnyLw4/PF7wXMAWrPMXo=";
-        revert = true;
+        url = "https://github.com/nodejs/node/commit/cd685fe3b6b18d2a1433f2635470513896faebe6.patch?full_index=1";
```

Comment:
Should `?full_index=1` always be appended to the URL when pulling patches from GitHub?

I wonder of omitting it can cause issues similar to what @ruuda found in https://github.com/NixOS/nixpkgs/pull/421152#issuecomment-3020344032

Reply:
https://stackoverflow.com/a/18134919

> This question is actually answered in [Chapter 7 of the Pro Git book](http://git-scm.com/book/en/v2/Git-Tools-Revision-Selection#Short-SHA-1):
>
>> Generally, eight to ten characters are more than enough to be unique within a project. For example, as of February 2019, the Linux kernel (which is a fairly sizable project) has over 875,000 commits and almost seven million objects in its object database, with no two objects whose SHA-1s are identical in the first 12 characters.